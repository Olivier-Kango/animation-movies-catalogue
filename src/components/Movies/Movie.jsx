import React from 'react';

const Movie = (movie) => {
  const { image, title } = movie;
  return <img src={`https://image.tmdb.org/t/p/w300${image}`} alt={`${title}`} />;
};

export default Movie;
