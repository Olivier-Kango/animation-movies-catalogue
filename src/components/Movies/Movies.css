.navbar {
  display: flex;
  justify-content: center;
  position: static;
  background: white;
  padding-top: 10px;
  width: 100%;
}