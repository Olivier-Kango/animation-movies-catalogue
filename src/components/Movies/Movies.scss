$color: #4369b2;

.movies {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  gap: 5px;

  div {
    border: 1px solid rgba(255, 255, 255, 0.112);
    box-shadow: 1px 1px 10px rgba(255, 255, 255, 0.123);
    border-radius: 7px;
    transform-style: preserve-3d;
    overflow: hidden;
    background: $color;

    img {
      width: 350px;
      margin: 0;
      border-radius: 7px 7px 0 0;
      transition: all 0.5s ease-in-out;
    }

    p {
      font-family: 'Lato', sans-serif;
      margin: 0 0 5px 0;
      color: rgb(255, 255, 255);
    }
  }

  div img:hover {
    transform: scale(1.04) translateY(-11px);
    border-radius: 0;
  }
}
