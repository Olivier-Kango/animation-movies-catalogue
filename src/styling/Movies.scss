$color: #4369b2;

.movies {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  gap: 5px;

  div {
    border: 1px solid rgba(255, 255, 255, 0.112);
    box-shadow: 1px 1px 10px rgba(255, 255, 255, 0.123);
    border-radius: 7px;
    transform-style: preserve-3d;
    transition: all 0.5s ease-in-out;
    background: $color;

    img {
      width: 350px;
      margin: 0;
      border-radius: 7px 7px 0 0;
    }

    p {
      font-family: 'Lato', sans-serif;
      margin: 0 0 5px 0;
      color: rgb(255, 255, 255);
    }
  }

  div:hover {
    transform: scale(1.2);
    border: 1.5px solid rgba(255, 255, 255, 0.345);
    box-shadow: 3px 3px 10px rgba(255, 255, 255, 0.345);
    z-index: 2;
  }
}
