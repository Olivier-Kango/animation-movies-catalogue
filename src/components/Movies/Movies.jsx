import React, { useEffect, useRef, useState } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMovies } from '../../redux/movies/movies';
import Movie from './Movie';
import YearsFilter from '../Filter/Filter';
import './Movies.css';

const Movies = () => {
  const moviess = useSelector((state) => state.movies);
  const dispatch = useDispatch();
  useEffect(() => { if (moviess.length === 0) { dispatch(fetchMovies('Home')); } }, []);

  const moviesAll = Array.from(moviess).sort((a, b) => b.popularity - a.popularity);
  const [filterTextValue, updateFilterText] = useState('All');

  const movies = moviesAll.filter((movie) => {
    if (filterTextValue === 'All') {
      return movie;
    }
    return filterTextValue === movie.release_date.split('-')[0];
  });

  const ref = useRef(null);
  const handleClick = (e) => {
    e.preventDefault();
    dispatch(fetchMovies(ref.current.value));
  };

  const onFilterValueSelected = (filterValue) => {
    updateFilterText(filterValue);
  };

  return (
    <div className="container">
      <h1>Movies Catalogue</h1>
      <form onSubmit={handleClick}>
        <input
          type="text"
          id="search"
          name="search"
          ref={ref}
        />
        <button
          type="submit"
          id="button-search"
        >
          Search
        </button>
      </form>
      <YearsFilter loadedMovies={moviesAll} filterValueSelected={onFilterValueSelected} />
      <br className="br" />
      <div className="movies">
        {movies.map((movie) => (
          <div
            key={`${movie.id}`}
            id={movie.id}
            className="movie"
          >
            <Link to={`/${movie.id}`} className="link">
              <Movie
                image={movie.poster_path}
                id={movie.id}
                key={`${movie.id}`}
              />
            </Link>
            <p>
              Popularity:&nbsp;
              {movie.popularity}
            </p>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Movies;
