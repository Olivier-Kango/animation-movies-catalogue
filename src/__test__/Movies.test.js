import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
  Routes,
  Route,
} from 'react-router-dom';
import Movies from '../components/Movies/Movies';
import store from '../redux/configureStore';

describe('Testing the Movies page:', () => {
  test('Should render', () => {
    render(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={<Movies />} />
          </Routes>
        </Router>
      </Provider>,
    );
  });

  it('fectch Movie from API', () => {
    const { container } = render(
      <Router>
        <Routes>
          <Route path="/:id" element={<Movies />} />
        </Routes>
      </Router>,
    );
    container.querySelector('.movie');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Display Movies from API', () => {
    const { container } = render(
      <Router>
        <Routes>
          <Route path="/:id" element={<Movies />} />
        </Routes>
      </Router>,
    );
    container.querySelector('.movies');
    expect(container.firstChild).toMatchSnapshot();
  });
});
