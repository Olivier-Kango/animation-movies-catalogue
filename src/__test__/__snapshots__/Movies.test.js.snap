// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Movies page: Display Movies from API 1`] = `null`;

exports[`Testing the Movies page: fectch Movie from API 1`] = `null`;
